src/transformers/models/pop2piano/feature_extraction_pop2piano.py

```diff
@@ -0,0 +1,288 @@
+
+        return bpm, beat_times, confidence, estimates, essentia_beat_intervals
+
+    def interpolate_beat_times(self, beat_times, steps_per_beat, extend=False):
+        beat_times_function = scipy.interpolate.interp1d(
+            np.arange(beat_times.size),
+            beat_times,
+            bounds_error=False,
+            fill_value="extrapolate",
+        )
+        if extend:
+            beat_steps_8th = beat_times_function(np.linspace(0, beat_times.size, beat_times.size * steps_per_beat + 1))
+        else:
+            beat_steps_8th = beat_times_function(
+                np.linspace(0, beat_times.size - 1, beat_times.size * steps_per_beat - 1)
+            )
+        return beat_steps_8th
+
+    def extrapolate_beat_times(self, beat_times, n_extend=1):
+        beat_times_function = scipy.interpolate.interp1d(
+            np.arange(beat_times.size),
+            beat_times,
+            bounds_error=False,
+            fill_value="extrapolate",
+        )
+        ext_beats = beat_times_function(np.linspace(0, beat_times.size + n_extend - 1, beat_times.size + n_extend))
+
+        return ext_beats
+
+    def preprocess_mel(
+        self,
+        audio,
+        beatstep,
+        n_bars,
+        padding_value,
+    ):
+        """Preprocessing for log-mel spectrogram"""
+
+        n_steps = n_bars * 4
+        n_target_step = len(beatstep)
+        ext_beatstep = self.extrapolate_beat_times(beatstep, (n_bars + 1) * 4 + 1)
+
+        batch = []
+        for i in range(0, n_target_step, n_steps):
+            start_idx = i
+            end_idx = min(i + n_steps, n_target_step)
+
+            start_sample = int(ext_beatstep[start_idx] * self.sampling_rate)
+            end_sample = int(ext_beatstep[end_idx] * self.sampling_rate)
+            feature = audio[start_sample:end_sample]
+            batch.append(feature)
+        batch = pad_sequence(batch, batch_first=True, padding_value=padding_value)
+
+        return batch, ext_beatstep
+
+    def single_preprocess(
+        self,
+        beatstep,
+        audio=None,
+    ):
+        """Preprocessing method for a single sequence."""
+
+        if audio is not None:
+            if len(audio.shape) != 1:
+                raise ValueError(
+                    f"Expected `audio` to be a single channel audio input of shape to be (n, ) but found {audio.shape}."
+                )
+
+        if beatstep[0] > 0.01:
+            logger.warning(f"`beatstep[0]` is not 0. All beatstep will be shifted by {beatstep[0]}.")
+            beatstep = beatstep - beatstep[0]
+
+        batch, ext_beatstep = self.preprocess_mel(
+            audio,
+            beatstep,
+            n_bars=self.n_bars,
+            padding_value=self.padding_value,
+        )
+
+        return batch, ext_beatstep
+
+    def __call__(
+        self,
+        raw_audio: Union[np.ndarray, List[float], List[np.ndarray]],
```

Comment:
I suggest naming this `audio`. In other models we have the warning: "Using `raw_speech` as a keyword argument is deprecated. Use `audio` instead." So might as well drop the "raw" here too.